### PR TITLE
test(backend): fix open handles caused by redis

### PR DESCRIPTION
**Pull request by stanleynguyen**
**Note**: pending #1360 to be merged first

- problem: currently we're initializing redis clients in the main
program of redis.service, which hoist the variables in the global
scope through importing. As a result, the file handle opened by
`createClient` function call doesn't get closed when the application
scope exits

- solution: have a dedicated init function inside redis.service so
we will initialize redis clients in the application scope specifically

- some misc changes:
  - update of sequelize version to latest
  - deprecate body-parser in favor of native express middlewares

### CODE REVIEW

File: backend/src/email/middlewares/tests/email.middleware.test.ts [hunk 1]
```diff
@@ -21,22 +21,25 @@ beforeEach(() => {
...
+  } as CreationAttributes<Campaign>)
   campaignId = campaign.id
 })
 
 afterAll(async () => {
   await Campaign.destroy({ where: {} })
   await User.destroy({ where: {} })
   await sequelize.close()
-  await RedisService.shutdown()
+  // await RedisService.shutdown()
```
Comment (prestonlimlianjie) [inline backend/src/email/middlewares/tests/email.middleware.test.ts:42]: As you pointed out in our offline discussion, we can remove the commented out lines

File: backend/src/email/routes/tests/email-transactional.routes.test.ts [hunk 2]
```diff
@@ -36,7 +37,7 @@ afterAll(async () => {
   await User.destroy({ where: {} })
   await sequelize.close()
 
-  await new Promise((resolve) => RedisService.rateLimitClient.flushdb(resolve))
+  await new Promise((resolve) => RedisService.rateLimitClient?.flushdb(resolve))
   await RedisService.shutdown()
```
Comment (prestonlimlianjie) [inline backend/src/email/routes/tests/email-transactional.routes.test.ts:41]: Should be deleted as well

**Review by prestonlimlianjie — APPROVED** [review]
lgtm apart from the 2 nits

**Comment by stanleynguyen** [comment]
Circling back on this @prestonlimlianjie, it seems that this error (attached below) seems to crash the program out of the open handles, hence "fixing" the open handles. But once I fixed the attached error, the open handles from redis seems to come back. I'm investigating further

Screenshot 2022-02-08 at 5 15 01 PM

**Comment by stanleynguyen** [comment]
Open handle when redis is completely removed

Screenshot 2022-02-08 at 5 33 02 PM

**Pull request merged**